presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -147,58 +149,48 @@ protected InternalResourceGroup(Optional<InternalResourceGroup> parent, String n
-                    runningTimeLimit,
+                    hardConcurrencyLimit,
                     maxQueuedQueries,
+                    runningTimeLimit,
                     queuedTimeLimit,
-                    getState(),
-                    eligibleSubGroups.size(),
                     DataSize.succinctBytes(cachedMemoryUsageBytes),
-                    runningQueries.size() + descendantRunningQueries,
                     queuedQueries.size() + descendantQueuedQueries,
-                    infos);
+                    eligibleSubGroups.size(),
+                    subGroups.values().stream()
+                            .map(InternalResourceGroup::getSummaryInfo)
+                            .collect(toImmutableList()),
+                    getAggregatedRunningQueriesInfo(),
+                    getPathToRoot());
         }
     }
 
-    public ResourceGroupStateInfo getStateInfo()
+    private ResourceGroupInfo getSummaryInfo()
     {
         synchronized (root) {
-            return new ResourceGroupStateInfo(
+            return summaryResourceGroupInfo(
                     id,
                     getState(),
+                    schedulingPolicy,
+                    schedulingWeight,
                     DataSize.succinctBytes(softMemoryLimitBytes),
-                    DataSize.succinctBytes(cachedMemoryUsageBytes),
                     softConcurrencyLimit,
                     hardConcurrencyLimit,
                     maxQueuedQueries,
                     runningTimeLimit,
                     queuedTimeLimit,
-                    getAggregatedRunningQueriesInfo(),
+                    DataSize.succinctBytes(cachedMemoryUsageBytes),
                     queuedQueries.size() + descendantQueuedQueries,
+                    runningQueries.size() + descendantRunningQueries,
+                    eligibleSubGroups.size(),
                     subGroups.values().stream()
-                            .map(subGroup -> new ResourceGroupInfo(
-                                    subGroup.getId(),
-                                    DataSize.succinctBytes(subGroup.softMemoryLimitBytes),
-                                    subGroup.softConcurrencyLimit,
-                                    subGroup.hardConcurrencyLimit,
-                                    subGroup.runningTimeLimit,
-                                    subGroup.maxQueuedQueries,
-                                    subGroup.queuedTimeLimit,
-                                    subGroup.getState(),
-                                    subGroup.eligibleSubGroups.size(),
-                                    DataSize.succinctBytes(subGroup.cachedMemoryUsageBytes),
-                                    subGroup.runningQueries.size() + subGroup.descendantRunningQueries,
-                                    subGroup.queuedQueries.size() + subGroup.descendantQueuedQueries))
+                            .map(InternalResourceGroup::getSummaryInfo)
```

Comment:
This is another logical difference of the new version from the previous version. Previously, the `ResourceGroupInfo` in `pathToRoot` does not have sub group info at all.

Reply:
Yes, intentional.